### DISCUSSION TITLE
Channel send on MediaStreamTrack?

### DISCUSSION THREAD

**Discussion by sabuone** [aiortc/aiortc#603] [Q&A]
I'm trying to channel send to return the value to the user after frame analysis in the media streamtrack's recv, 

but I don't know how to do it.

How do you take care of it?

**Comment by bmsan** [comment]
@sabuone when you say "to return the value" what does "the value" represent?
If it is represented by statistics/metadata you can use the data channel.
Did you take a look at the [server example?](https://github.com/aiortc/aiortc/blob/main/examples/server/server.py) it uses both a data channel and a video channel.

**Comment by jlaine** [comment]
Thanks @bmsan, the server example is the right place to start. I'm converting this to a discussion.

**Comment by zhongyi-zhou** [comment]
@jlaine Hi! I think I face a similar problem. Similar to the "server" example, I need to process each frame and not only return a frame but also some JSON data. For example, given each frame, I want to run an AI app at the backend and return some classification results in real time with a visualization output (which is an image). The "server" example told us how to send the visualized image back but I am not sure how to send the JSON data with the visualized image.

data channel might work, but from my understanding, it is more for a chatting use scenario. I am not quite how to utilize this data channel in my case since the chatting track is not synced with the video frame processing.
